crates/wasmtime/src/runtime/component/resource_table.rs

```diff
@@ -100,18 +102,53 @@ impl TableEntry {
 impl ResourceTable {
     /// Create an empty table
     pub fn new() -> Self {
-        ResourceTable {
+        let mut me = ResourceTable {
             entries: Vec::new(),
             free_head: None,
-        }
+            debug: false,
+        };
+
+        // Reserve 0 as an invalid entry.  This effectively reserves
+        // `Resource::new_own(0)` as a sentinal value meaning "this resource
+        // doesn't exist anymore".
+        me.push(Tombstone).unwrap();
```

Comment:
While this is true of indices we hand out to wasm itself, I think this can be removed now? I think that this was required before merging the wasm-facing tables together but nowadays I think this should be unnecessary